src/container_type.rs

```diff
@@ -0,0 +1,84 @@
+use k8s_openapi::{
+    api::core::v1::ResourceRequirements, apimachinery::pkg::api::resource::Quantity,
+};
+
+/// Describes the type of container. The types are:
+///
+/// - `Main`: This type of container runs the main appliction, like Superset or
+///    NiFi
+/// - `Init`: This type of container runs only once at the start of the product
+///    cluster and executes initial tasks like database migrations or insertion
+///    of data
+/// - `Sidecar`: Sidecar containers run alongside the main containers for
+///    additional tasks
+#[derive(Clone, Debug, Default)]
```

Comment:
If we go with the ContainerType I would make it more explicit, e.g. have
* Product (main)
* copy files (init)
* text replacement (init)
* create key and truststore (init)
* logging (sidecar)
* metrics (sidecar)
* healthcheck (sidecar)
* ...

Reply:
The exact variants should be discussed in the upcoming separate PR. And yes we can be more explicit about the operations the container executes.